pkg/util/util.go

```diff
@@ -45,6 +47,71 @@ func GenerateRandomString(n int) string {
 	return string(b)
 }
 
+// StringInSlice returns if a passed string exists in the passed slice of strings
+func StringInSlice(checkStr string, strSlice []string) bool {
+	// Iterate the slice
+	for _, ele := range strSlice {
+		// if current element is same as element to search(checkStr), return true indicating checkStr found
+		if ele == checkStr {
+			return true
+		}
+	}
+	// Exited loop without finding required element(checkStr) in slice, so return false indicating checkStr not found in slice
+	return false
+}
+
+// GetRandomName returns a randomly generated name which can be used for naming odo and/or openshift entities
+// prefix: Desired prefix part of the name
+// existList: List to verify that the returned name does not already exist
+// preferredSuffix: Optional suffix if passed, will be checked if prefix-suffix does not exist in existList if not, will add an additional timestamp to make name unique
+// Returns randomname is prefix-suffix, if suffix is passed else generated. Aditionally, if the prefix-suffix is used, it'll be appended with additional 4 char string to take the form prefix-suffix-{a-z A-z}4+
+func GetRandomName(prefix string, existList []string, preferredSuffix string) string {
+	if preferredSuffix == "" {
+		// Generate suffix if not passed, using random country names generated from Pallinder/go-randomdata as suffix
+		preferredSuffix = strings.Replace(
+			strings.Replace(strings.Replace(
+				strings.Replace(
+					strings.Replace(
+						strings.ToLower(randomdata.Country(randomdata.FullCountry)),
+						" ",
+						"-",
+						-1,
+					),
+					".",
+					"-",
+					-1,
+				),
+				",",
+				"-",
+				-1,
+			),
+				"(",
+				"-",
+				-1,
+			),
+			")",
+			"-",
+			-1,
+		)
+	}
+	// name is prefix-suffix
+	name := fmt.Sprintf("%s-%s", prefix, preferredSuffix)
+	// check if generated name is already used in the existList
+	if StringInSlice(name, existList) {
+		prevName := name
+		// keep generating names until generated name is not unique. So, loop terminates when name is unique and hence for condition is false
+		for StringInSlice(prevName, existList) {
```

Comment:
A max number of retries would a nice addition to prevent it from a infinite loop in worst case scenarios.

Reply:
Agreed it would be safe

Reply:
will add it